plugins/workunitservices/workunitservices.ipp

```diff
@@ -78,13 +114,27 @@ inline bool serializeWUSrow(IPropertyTree &pt,MemoryBuffer &mb, bool isonline)
     short int prioritylevel = calcPriorityValue(&pt);
     mb.appendEndian(sizeof(prioritylevel), &prioritylevel);
 
-    const char *mod = "TimeStamps/TimeStamp[@application=\"workunit\"]/Modified";
-    const char *crt = "TimeStamps/TimeStamp[@application=\"workunit\"]/Created";
-    fixedAppend(mb,20,pt,crt);
-    if (pt.hasProp(mod))
-        fixedAppend(mb,20,pt,mod);
+    StringBuffer s1, s2;
+    const char* modTime = readModifyTime(pt, s1, true);
+    const char* crtTime = readCreateTime(pt, s2, true);
+    if (crtTime && *crtTime)
```

Comment:
Minor:  Possibly simpler to not return a char *, and use the buffer passed in instead.  E.g., if (crtTime.length())